### PR TITLE
Convert and quote file names for BatchUnstageFiles

### LINKED ISSUE #7607: Unstage file not doing anything due to missing quotes when Git Reset file
## Current behaviour

Unstage file button is not doing anything when file path contains spaces.

## Expected behaviour

The Unstage button in the commit window should remove the selected file from the staged list

## Steps to reproduce

1) Create a git repo (tested on Windows)
2) Commit a file into a folder containing spaces (probably fails also if the file contains spaces instead of the folder)
3) Modify that file inside a folder containing spaces
4) Stage the file in the Commit window
5) Try to Unstage that single file (the Unstage all button does work, but not if trying to do file by file)

## Screenshots

2020-01-07 Git Extensions - Unstage defect - missing quotes in reset file

## Did this work in previous version of GitExtensions

I don't know

## Environment

- Git Extensions 3.3.1.7897
- Build 5a97671645532bcedc443bac7b727f40db47cb5c
- Git 2.24.1.windows.2
- Microsoft Windows NT 10.0.17134.0
- .NET Framework 4.8.4075.0
- DPI 96dpi (no scaling)

## Diagnostics

`Command log` shows that the Git Reset command is it invoking is not wrapped in quotes (so git is not considering as a single file path, and is not matching the file passed)

### PR DESCRIPTION
**Pull request by mstv**
Fixes #7607

## Proposed changes

- Convert and quote file names for `GitModule.BatchUnstageFiles`

## Test methodology 

- manual
- adapted NUnit tests

## Test environment(s) 

- Git Extensions 3.3.0
- Build 38d5f9135c7cb0099e42c58cbe32f6c5b96a78e8
- Git 2.24.1.windows.2
- Microsoft Windows NT 6.2.9200.0
- .NET Framework 4.8.4075.0
- DPI 96dpi (no scaling)

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitCommands/ArgumentBuilderExtensions.cs [hunk 1]
```diff
@@ -379,5 +379,8 @@ public static List<BatchArgumentItem> BuildBatchArguments(this ArgumentBuilder b
 
             return batches;
         }
+
+        public static List<BatchArgumentItem> BuildBatchArgumentsForFiles(this ArgumentBuilder builder, IEnumerable<string> files)
+            => builder.BuildBatchArguments(files.Select(f => f.ToPosixPath().Quote()));
```
[the remark this answers is not in the document]
Comment (mstv) [reply GitCommands/ArgumentBuilderExtensions.cs]: > Suggest to use .QuoteNE()

yep, done and rebased

**Comment by mstv** [comment]
@msftbot merge in 1 day

**Comment by vbjay** [comment]
Can this be patched on the 3.3 branch and released?  I saw this on 

- Git Extensions 3.3.1.7897
- Build 5a97671645532bcedc443bac7b727f40db47cb5c
- Git 2.25.0.windows.1
- Microsoft Windows NT 10.0.18363.0
- .NET Framework 4.8.4150.0
- DPI 96dpi (no scaling)

**Comment by RussKie** [comment]
We're getting close to releasing v4. Making a new release is an involved process.

**Pull request merged**